src/tools/tidy/src/deps.rs

```diff
@@ -496,6 +500,8 @@ const PERMITTED_STDLIB_DEPENDENCIES: &[&str] = &[
     "windows_x86_64_gnu",
     "windows_x86_64_gnullvm",
     "windows_x86_64_msvc",
+    "zerocopy",
+    "zerocopy-derive",
```

Comment:
Where is zerocopy-derive coming from? We have default features off for `rand` so it shouldn't be through there, I'm not seeing it pop up anywhere in `cargo tree` for the library.

`rand` is only a dev dependency so it doesn't really matter, the extra items are just surprising.

Reply:
I'm guessing it is getting confused by this unusual pinning approach in zerocopy:

```
[target."cfg(any())".dependencies.zerocopy-derive]
version = "=0.8.17"
```

The dependency list here unconditionally includes all target dependencies.

Reply:
Huh, that is interesting. The deps appear in `library/Cargo.lock` though, I guess the pinning causes them to show up there even if they aren't used?

Reply:
Yea, cargo does the same thing. `Cargo.lock` is generated as if all the `[target]` tables are included unconditionally.

I don't think I've ever seen someone use this approach of `cfg(any())`, it's quite interesting.